// @alifd/meet-react@2.9.7/es/select/section/select-value.js
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _defineProperty(obj, key, value) { key = _toPropertyKey(key); if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return _typeof(key) === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (_typeof(input) !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (_typeof(res) !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); }
import React, { createElement, useRef, useContext } from "react";
import classNames from 'classnames';
import { ItemContext } from '../../form';
import TagSelect from '../../tag-select';
import View from '../../view';
import Tag from '../../tag';
import { guid } from '../../utils/hooks';
var SelectValue = function SelectValue(props) {
  var _classNames, _classNames2;
  var items = props.items,
    prefix = props.prefix,
    displayType = props.displayType,
    sizeProp = props.size,
    disabled = props.disabled,
    type = props.type;
  var _useContext = useContext(ItemContext),
    isInsideForm = _useContext.isInsideForm,
    labelAlign = _useContext.labelAlign,
    contextSize = _useContext.size;
  var inlineMode = displayType === 'tag' && isInsideForm && labelAlign === 'inset';
  var clsPrefix = "".concat(prefix, "select");
  var isInverse = type === 'inverse';
  var size = sizeProp || (isInsideForm ? contextSize : 'medium');
  var ref = useRef(guid('select-val-'));
  switch (displayType) {
    case 'tag':
      return /*#__PURE__*/React.createElement(TagSelect, {
        disabled: disabled,
        isPreview: true,
        inlineMode: inlineMode,
        prefix: prefix,
        size: size,
        dataSource: items
      });
    case 'tag-value':
      return /*#__PURE__*/React.createElement(View, {
        className: classNames("".concat(clsPrefix, "-tag-value"), (_classNames = {}, _defineProperty(_classNames, "".concat(clsPrefix, "-value--inverse"), isInverse), _defineProperty(_classNames, "".concat(clsPrefix, "-value--disabled"), disabled), _defineProperty(_classNames, "".concat(clsPrefix, "-value--inverse-disabled"), disabled && isInverse), _classNames))
      }, items.map(function (val) {
        return /*#__PURE__*/React.createElement(Tag, {
          key: "".concat(ref.current, "-").concat(val),
          className: "".concat(clsPrefix, "-tag-item"),
          type: "primary",
          size: size
        }, "".concat(val));
      }));
    default:
      return /*#__PURE__*/React.createElement(View, {
        className: classNames((_classNames2 = {}, _defineProperty(_classNames2, "".concat(clsPrefix, "-value--inverse"), isInverse), _defineProperty(_classNames2, "".concat(clsPrefix, "-value--disabled"), disabled), _defineProperty(_classNames2, "".concat(clsPrefix, "-value--inverse-disabled"), disabled && isInverse), _classNames2))
      }, items.join(','));
  }
};
export default SelectValue;
